src/audio/copier/copier.c

```diff
@@ -1625,8 +1670,26 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 		buffer_release(source_c);
 	}
 
+	/* update params for the DMA buffer */
+	switch (dev->ipc_config.type) {
+	case SOF_COMP_HOST:
+		if (cd->ipc_gtw || params->direction == SOF_IPC_STREAM_PLAYBACK)
+			break;
+		COMPILER_FALLTHROUGH;
+	case SOF_COMP_DAI:
+		if (cd->endpoint_num > 1 || params->direction == SOF_IPC_STREAM_CAPTURE)
```

Comment:
aah, ouch, sorry, my bad! The above should certainly be
```
if (dev->ipc_config.type == SOF_COMP_DAI &&
    (cd->endpoint_num > 1 || params->direction == SOF_IPC_STREAM_CAPTURE))
```

Reply:
@lyak I think it's not needed because for the host endpoint _num can never be > 1

Reply:
but `params->direction == SOF_IPC_STREAM_CAPTURE` will be true, won't it?

Reply:
ahh yes. fixed